cmd/apexctl/device.go

```diff
@@ -0,0 +1,106 @@
+package main
+
+import (
+	"encoding/json"
+	"fmt"
+	"log"
+
+	"github.com/google/uuid"
+	"github.com/redhat-et/apex/internal/client"
+	"github.com/redhat-et/apex/internal/models"
+	"github.com/urfave/cli/v2"
+)
+
+func listAllDevices(cCtx *cli.Context) error {
+	c, err := client.NewClient(cCtx.Context,
+		cCtx.String("host"),
+		client.WithPasswordGrant(
+			cCtx.String("username"),
+			cCtx.String("password"),
+		),
+	)
+
+	if err != nil {
+		log.Fatal(err)
+	}
+
+	allPeers := getAllPeers(c)
+
+	switch cCtx.String("output") {
+	case encodeJsonPretty:
+		var devices []models.Device
```

Comment:
If we can, it would be cool to make this function generic.
e.g
```go
func FormatOutput(result interface{}) (string, error)
```

That should work for JSON but I'm not sure about the tabwriter.